R/calcFEShares.R

```diff
@@ -2,28 +2,32 @@
 #'
 #' @param subtype 'ind_coal' for the share of coal used in industry. 'ind_bio' for the share of biomass used in industry
 #'
-#' @author Antoine Levesque
 calcFEShares <- function(subtype) {
-
-  #---- Read Data -----
+  if (! subtype %in% c("ind_coal", "ind_bio", "ind_liq")) {
+    stop("Unknown subtype.")
+  }
 
   edge_buildings <- calcOutput("IOEdgeBuildings", subtype = "output_EDGE_buildings", aggregate = FALSE)
   output <- calcOutput("IO", subtype = "output", aggregate = FALSE)
-  fe_demand <- calcOutput("FEdemand", aggregate = FALSE)[, , "SSP1"]
+  fe_demand <- calcOutput("FEdemand", scenario = "SSP1", aggregate = FALSE)
```

Comment:
Disables the `madrat` cache for this call.  See https://github.com/pik-piam/mrindustry/pull/30#discussion_r1965132521 and https://github.com/pik-piam/mrindustry/pull/30#discussion_r1965141924.  Should be fixed for the benefit of run times and disk space.  And is probably an issue across related changes in other packages.

Reply:
You mean it doesn't use a potential cache, in which the SSP1 scenario could be available right? I don't see a way around this without keeping a "default" set of scenarios, which is exactly what we want to get rid of.

Reply:
`calcFEShares()` is called by `fullREMIND()`, and that knows what scenarios are going to be calculated.  Just pass the `scenario` [sic] parameter passed to e.g. `calcFEdemand()` to `calcFEshares()` too.

Reply:
Ok! Thanks, will implement this right away.

Reply:
Thank you for pointing this out, Michaja.  I just tested and noticed that madrat even does not recognize when you pass the same scenarios but in different order and will recalculate.

Reply:
> I just tested and noticed that madrat even does not recognize when you pass the same scenarios but in different order and will recalculate

:point_right: https://github.com/pik-piam/mrindustry/pull/30#discussion_r1965141924

Reply:
Ah, cool! From what I can tell, @johanneskoch94 has adapted this practice in this PR as well :+1: